From a36e994704b6953cf88c6031b9774c62dc45b5d9 Mon Sep 17 00:00:00 2001
From: Enrico Berti <hello@enricoberti.com>
Date: Thu, 9 Apr 2015 13:42:07 +0200
Subject: [PATCH 1153/1173] [filebrowser] Fixes modal upload in IE9 and IE10

Added CSRF hidden input for the iframe fallback
Fixed check on __proto__
---
 .../desktop/static/desktop/ext/js/fileuploader.js  |   12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/desktop/core/src/desktop/static/desktop/ext/js/fileuploader.js b/desktop/core/src/desktop/static/desktop/ext/js/fileuploader.js
index da2f00f..45bb34c 100644
--- a/desktop/core/src/desktop/static/desktop/ext/js/fileuploader.js
+++ b/desktop/core/src/desktop/static/desktop/ext/js/fileuploader.js
@@ -1042,6 +1042,12 @@ qq.extend(qq.UploadHandlerForm.prototype, {
         dest.value = params.dest;
         form.appendChild(dest);
 
+        var csrfmiddlewaretoken = document.createElement('input');
+        csrfmiddlewaretoken.type = 'hidden';
+        csrfmiddlewaretoken.name = 'csrfmiddlewaretoken';
+        csrfmiddlewaretoken.value = $.cookie('csrftoken');
+        form.appendChild(csrfmiddlewaretoken);
+
         var self = this;
         this._attachLoadEvent(iframe, function(){
             self.log('iframe loaded');
@@ -1095,10 +1101,10 @@ qq.extend(qq.UploadHandlerForm.prototype, {
             response;
 
         this.log("converting iframe's innerHTML to JSON");
-        this.log("innerHTML = " + doc.body.innerHTML);
+        this.log("innerHTML = " + $(doc.body.innerHTML).text());
 
         try {
-            response = eval("(" + doc.body.innerHTML + ")");
+            response = eval("(" + $(doc.body.innerHTML).text() + ")");
         } catch(err){
             response = {};
         }
@@ -1182,7 +1188,7 @@ qq.extend(qq.UploadHandlerXhr.prototype, {
         // HUE-815: [fb] Upload button does not work in Firefox 3.6
         // see https://github.com/valums/ajax-upload/issues/91
         //if (!(file instanceof File)){
-        if (!(file instanceof File || file.__proto__.constructor.name == 'File' || file instanceof Object) ){
+        if (!(file instanceof File || (file.__proto__ && file.__proto__.constructor.name == 'File') || file instanceof Object) ){
             throw new Error('Passed obj in not a File (in qq.UploadHandlerXhr)');
         }
 
-- 
1.7.9.5

